google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQuery.java

```diff
@@ -440,6 +443,72 @@ public static QueryResultsOption maxWaitTime(long maxWaitTime) {
     }
   }
 
+  class QueryOption implements Serializable {
+    private static final long serialVersionUID = 6206193419355824689L;
+
+    private final Object option;
+
+    private QueryOption(Object option) {
+      this.option = option;
+    }
+
+    public QueryResultsOption getQueryResultsOption() {
+      return option instanceof QueryResultsOption ? (QueryResultsOption) option : null;
+    }
+
+    public RetryOption getRetryOption() {
+      return option instanceof RetryOption ? (RetryOption) option : null;
+    }
+
+    public static QueryResultsOption[] filterQueryResutlsOptions(QueryOption... options) {
```

Comment:
spelling (Resutls)

Also why is it necessary for this to be public?

Reply:
Fixed the typo. It doesn't have to be public, but I thought it would not hurt to be public. Changed to package-private.